packages/app/src/systems/Ecosystem/machines/ecosystemMachine.tsx

```diff
@@ -0,0 +1,235 @@
+import type { Project } from '../types';
+
+import { FetchMachine } from '~/systems/Core';
+
+export type EcosystemInputs = {
+  filter: {
+    tag?: string;
+    projects?: Project[];
+  };
+  search: {
+    query?: string;
+    projects?: Project[];
+  };
+};
+
+type MachineContext = {
+  projects: Project[];
+  filteredProjects: Project[];
+  search: string;
+  tags: string[];
+  filter: string;
+};
+
+type MachineServices = {
+  fetchProjectsAndTags: {
+    data: {
+      projects: Project[];
+      tags: string[];
+    };
+  };
+  searchProjects: {
+    data: Project[];
+  };
+  filterProjects: {
+    data: Project[];
+  };
+};
+
+type EcosystemMachineEvents =
+  | {
+      type: 'FILTER';
+      input: { tag?: string; projects?: Project[] };
+    }
+  | {
+      type: 'SEARCH';
+      input: { query?: string; projects?: Project[] };
+    }
+  | {
+      type: 'FETCH_PROJECTS_AND_TAGS';
+      input: null;
+    }
+  | {
+      type: 'CLEAR_FILTER';
+      input: null;
+    };
+
+const initialState: MachineContext = {
+  projects: [],
+  filteredProjects: [],
+  search: '',
+  tags: [],
+  filter: '',
+};
+export const ecosystemMachine = createMachine(
+  {
+    // eslint-disable-next-line @typescript-eslint/consistent-type-imports
+    tsTypes: {} as import('./ecosystemMachine.typegen').Typegen0,
+    schema: {
+      context: initialState as MachineContext,
+      services: {} as MachineServices,
+      events: {} as EcosystemMachineEvents,
+    },
+    predictableActionArguments: true,
+    id: '(machine)',
+    initial: 'idle',
+    states: {
+      idle: {
+        on: {
+          FETCH_PROJECTS_AND_TAGS: {
+            target: 'fetching',
+          },
+          FILTER: {
+            target: 'filtering',
+            actions: ['assignFilter'],
+          },
+          SEARCH: {
+            target: 'searching',
+            actions: ['assignSearch'],
+          },
+          CLEAR_FILTER: {
+            actions: ['clearFilter'],
+          },
+        },
+      },
+      fetching: {
+        tags: ['isLoading'],
+        invoke: {
+          src: 'fetchProjectsAndTags',
+          onDone: {
+            target: 'idle',
+            actions: ['assignProjectsAndTags', 'clearSearchAndFilter'],
+          },
+        },
+      },
+      filtering: {
+        entry: ['clearSearch'],
+        tags: ['isLoading'],
+        invoke: {
+          src: 'filterProjects',
+          data: {
+            input: (_: MachineContext, ev: EcosystemMachineEvents) => ({
+              ...ev.input,
+              projects: _.projects,
+            }),
+          },
+          onDone: {
+            target: 'idle',
+            actions: ['assignFilteredProjects'],
+          },
+        },
+      },
+      searching: {
+        entry: ['clearFilter'],
+        tags: ['isLoading'],
+        invoke: {
+          src: 'searchProjects',
+          data: {
+            input: (_: MachineContext, ev: EcosystemMachineEvents) => ({
```

Comment:
_ is used to skip lint when you don't wanna use the value of variable.. in this case can use `ctx`